web/src/components/data-entities/survey/SurveyList.tsx

```diff
@@ -0,0 +1,527 @@
+        flex: 1,
+        colId: OBSERVABLE_ITEM_ID_FIELD,
+      });
+
+    return cols;
+  }, []);
+
+  useEffect(() => {
+    document.title = 'Surveys';
+    // Empty array
+    cachedOptions.splice(0, cachedOptions.length);
+  }, []);
+
+  useEffect(() => {
+    let active = true;
+
+    if (!optionLoading) {
+      return undefined;
+    }
+
+    if (cachedOptions.length !== 0) {
+      // If user select more than two species, we will not show the dropdown with value such that user cannot enter
+      // more than two species
+      const f = gridRef?.current?.api.getFilterInstance(OBSERVABLE_ITEM_ID_FIELD);
+      if (!f?.getModel() || !f.getModel().operator) {
+        // operator attribute valid if filter on two species
+        setOptions(cachedOptions);
+      } else {
+        // Close the dropdown
+        setOpen(false);
+      }
+      return undefined;
+    }
+
+    (async () => {
+      const url = `reference/observableItems?pageSize=-1&sort=${encodeURIComponent('[{"field":"observation.name","order":"asc"}]')}`;
+
+      getResult(url)
+        .then((res) => {
+          res.data.items.forEach((i: Option) => cachedOptions.push(i));
+        })
+        .finally(() => {
+          if (active) {
+            setOptions(cachedOptions);
+          }
+        });
+    })();
+
+    return () => {
+      active = false;
+    };
+  }, [optionLoading]);
+
+  useEffect(() => {
+    if (!open) {
+      setOptions([]);
+    }
+  }, [open]);
+
+  // Create filter for species and push it to the ag grid to trigger page reload
+  const handleSpeciesFilterChange = useCallback((
+    event: React.SyntheticEvent,
+    value: Option[]
+  ) => {
+    const f = gridRef?.current?.api.getFilterInstance(OBSERVABLE_ITEM_ID_FIELD);
+
+    if (value.length === 0) {
+      (f?.setModel(null) as AgPromise<void>)
+        .then(() => gridRef?.current?.api.onFilterChanged());
+    } else if (value.length === 1) {
+      (f?.setModel({
+        filter: value[0].observableItemId,
+        filterType: 'text',
+        type: 'equals'
+      }) as AgPromise<void>).then(() => gridRef?.current?.api.onFilterChanged());
+    } else {
+
+      const model: ModelType = {
+        filterType: 'text',
+        operator: 'OR'
+      };
+
+      for (let i = 0; i < value.length; i++) {
+        model['condition' + (i + 1)] = {
+          filter: value[i].observableItemId,
+          filterType: 'text',
+          type: 'equals'
+        };
+      }
+
+      (f?.setModel(model) as AgPromise<void>).then(() => gridRef?.current?.api.onFilterChanged());
+    }
+  }, []);
+
+  const onGridReady = useCallback(
+    (event: GridReadyEvent) => {
+      async function fetchSurveys(e: GridReadyEvent) {
+        e.api.setDatasource({
+          // This is the functional structure need for datasource
+          rowCount: rowsPerPage,
+          getRows: (params) => {
+            let url = `data/surveys?page=${params.startRow / 100}`;
+            const conditions = [];
```

Comment:
there is 39 lines duplicate code from here (with `DiverList.tsx`).

Reply:
keep for now